src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```
@@ -0,0 +1,251 @@
+        )
+        arguments_str = " ".join(arguments)
+
+        # Set up docker run command
+        image = self.get_image(deployment=deployment)
+        docker_environment_str = " ".join(
+            f"-e {k}={v}" for k, v in environment.items()
+        )
+
+        start_time = time.time()
+
+        instance_type = settings.instance_type or self.DEFAULT_INSTANCE_TYPE
+
+        # Set up credentials
+        self.setup_credentials()
+
+        # Run the entire pipeline
+        try:
+            task = sky.Task(
+                run=f"docker run --rm {docker_environment_str} {image} {entrypoint_str} {arguments_str}",
+                setup=self.get_setup(stack),
+            )
+            task = task.set_resources(
+                sky.Resources(
+                    cloud=self.cloud,
+                    instance_type=instance_type,
+                    cpus=settings.cpus,
+                    memory=settings.memory,
+                    accelerators=settings.accelerators,
+                    accelerator_args=settings.accelerator_args,
+                    use_spot=settings.use_spot,
+                    spot_recovery=settings.spot_recovery,
+                    region=settings.region,
+                    zone=settings.zone,
+                    image_id=settings.image_id,
+                    disk_size=settings.disk_size,
+                    disk_tier=settings.disk_tier,
+                )
+            )
+
+            cluster_name = settings.cluster_name
+            if cluster_name is None:
+                # Find existing cluster
+                for i in sky.status(refresh=True):
+                    if isinstance(
+                        i["handle"].launched_resources.cloud, type(self.cloud)
+                    ):
+                        cluster_name = i["handle"].cluster_name
+                        logger.info(
+                            f"Found existing cluster {cluster_name}. Reusing..."
+                        )
```

Will `sky.launch` break if `cluster_name` is still `None`? I.e., do we need to raise an error here if no reusable cluster was found?

---

technically if no cluster was found it will create a new one and give it an auto generated name

---

yes itll create a new one